src/pages/index.js

```diff
@@ -140,39 +196,117 @@ const addCardPopup = new PopupWithForm(
 addCardPopup.setEventListeners();
 
 // Add Card Button
+const addCardButton = document.querySelector("#add-card-button");
+
 addCardButton.addEventListener("click", () => {
-  // Reset the Input fields on the add card popup
   addCardFormValidator.resetValidation();
 
   addCardPopup.open();
 });
 
+// Create a new card
 function handleCardSaveSubmit(inputValues) {
-  // Create a new card instance
-  const newCardInstance = renderCard({
+  const saveButton = document.getElementById("add-card-save-button");
+
+  saveButton.textContent = "Saving...";
+
+  const cardInfo = {
     name: inputValues.title,
     link: inputValues.url,
+  };
+
+  api
+    .addNewCard(cardInfo)
+    .then((res) => {
+      console.log("New Card:", cardInfo);
+      renderCard(res);
+    })
+    .catch((error) => {
+      console.error("API Error:", error);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      addCardPopup.close();
+    });
+}
+
+// *************************** Delete Card  ************************** //
+const confirmDeletePopup = new PopupWithSubmit({
+  popupSelector: "#delete-card-modal",
+});
+
+const confirmDeleteButton = document.querySelector(
+  "#confirm-delete-modal-button"
+);
+
+confirmDeletePopup.setEventListeners();
+
+function handleDeleteIconClick(card) {
+  confirmDeletePopup.open(card);
+  confirmDeleteButton.focus();
+  confirmDeletePopup.setSubmit(() => {
+    card._deleteCard();
+    confirmDeletePopup.close();
   });
 }
 
-// ************************ Inital Card Setup ************************ //
+// ************************** Preview Image ************************** //
+
+// Preview Image Popup
+const previewImagePopup = new PopupWithImage({
+  popupSelector: "#preview-image-modal",
+});
+
+// Clicking an Image to get the preview Image
+function handleImageClick(cardData) {
+  previewImagePopup.open(cardData);
+}
+
+previewImagePopup.setEventListeners();
+
+const renderCard = (card) => {
+  const cardInstance = new Card(
+    card,
+    "#card-template",
+    handleImageClick,
+    handleDeleteIconClick,
+    api
+  );
+
+  const cardElement = cardInstance.generateCard();
+  cardSection.addItem(cardElement);
+};
 
+// ************************* Validation ************************* //
+const formSettings = {
+  formSelector: ".modal__form",
+  inputSelector: ".modal__input",
+  submitButtonSelector: ".modal__save-button",
+  inactiveButtonClass: "modal__save-button_disabled",
+  inputErrorClass: "modal__input_type_error",
+  errorClass: "modal__error_visible",
+};
+
+const profileFormValidator = new FormValidator(formSettings, profileEditForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```